### PR TITLE
Apply user configuration to Istio reconciler

### PR DESCRIPTION
**Pull request by cnvergence**
See also: 
- https://github.com/kyma-incubator/reconciler/issues/1211

### CODE REVIEW

**Review by tobiscr — CHANGES_REQUESTED** [review]
@cnvergence : we have some collisions in the go.mod files.... otherwise, it looks good :)

File: pkg/reconciler/instances/istio/merge/merge.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+func applyIstioCR(istioCRList *v1alpha1.IstioList, operatorManifest string) (string, error) {
+	var outputManifest []byte
+	toBeInstalledIop := istioOperator.IstioOperator{}
+	err := json.Unmarshal([]byte(operatorManifest), &toBeInstalledIop)
+	if err != nil {
+		return "", err
+	}
+
+	for _, cr := range istioCRList.Items {
```
[the remark this answers is not in the document]
Comment (cnvergence) [reply pkg/reconciler/instances/istio/merge/merge.go:58]: Good question, I also thought about it, but I assumed that we would do some validation webhook for the IstioCR, to exclude the possibility of overwriting some of the configurations.
Other than that I don't see a problem with having more than one IstioCR.

File: pkg/reconciler/instances/istio/clientset/provider.go [hunk 2]
```diff
@@ -43,3 +50,35 @@ func (c *DefaultProvider) RetrieveFrom(kubeConfig string, log *zap.SugaredLogger
...
+	})
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+// restConfig loads the rest configuration needed by k8s clients to interact with clusters based on the kubeconfig.
+// Loading rules are based on standard defined kubernetes config loading.
+func restConfig(kubeconfigData string) (*rest.Config, error) {
```
[the remark this answers is not in the document]
Comment (cnvergence) [reply pkg/reconciler/instances/istio/clientset/provider.go:77]: In fact, we don't do it, the rename is a good suggestion

File: pkg/reconciler/instances/istio/merge/merge.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		return nil, err
+	}
+
+	return istioCRList, nil
+}
+
+func applyIstioCR(istioCRList *v1alpha1.IstioList, operatorManifest string) (string, error) {
+	var outputManifest []byte
+	toBeInstalledIop := istioOperator.IstioOperator{}
+	err := json.Unmarshal([]byte(operatorManifest), &toBeInstalledIop)
```
[the remark this answers is not in the document]
Comment (cnvergence) [reply pkg/reconciler/instances/istio/merge/merge.go:53]: As we have discussed, we will keep it here for now

**Review by triffer — DISMISSED** [review]

**Review by barchw — APPROVED** [review]

**Comment by cnvergence** [comment]
@tobiscr go.mod applied, please take a look now :)

**Review by barchw — DISMISSED** [review]
We need to configure default value for numTrustedProxies first

**Review by barchw — APPROVED** [review]
/lgtm

**Review by tobiscr — APPROVED** [review]

**Pull request merged**